### DISCUSSION TITLE
one interface followed with three retrieval, only the first retrieval component gets right input

### DISCUSSION THREAD

**Discussion by riverbutte** [infiniflow/ragflow#4114] [Q&A]
I create Agent from template, such as Text to SQL or DB Assistant, there are three retrieval components atfer one interface component, when running the agent, the first retrieval get user query as input, retrive right content, but the second retrieval component recieves the output of the first retrieval as input, but the three retrieval components are parallel. Why it happens, i set the input of retrieval componet as default.

**Comment by JinHai-CN** [comment]
Please refer to: https://ragflow.io/docs/dev/text2sql_agent
